### PR TITLE
Remove compatibility code not needed anymore

### PR DESCRIPTION
**Pull request by matusvalo**
This PR removes compatibility code supporting legacy Python versions. This PR does not introduce syntax changes so it should be compatible with cython -2. This PR is not interfering with #5810.

### CODE REVIEW

File: Cython/Compiler/Main.py [hunk 1]
```diff
@@ -9,15 +9,6 @@
 import sys
 import io
 
-if sys.version_info[:2] < (2, 7) or (3, 0) <= sys.version_info[:2] < (3, 3):
```
Comment (matusvalo) [inline Cython/Compiler/Main.py:12]: This line maybe should not be just deleted but adjusted to lowest supported version of python.

File: Cython/Build/Dependencies.py [hunk 2]
```diff
@@ -51,19 +51,8 @@
...
-        _fs_encoding = sys.getdefaultencoding()
-    def encode_filename_in_py2(filename):
-        if not isinstance(filename, bytes):
-            return filename.encode(_fs_encoding)
-        return filename
-else:
-    def encode_filename_in_py2(filename):
-        return filename
-    basestring = str
+def encode_filename_in_py2(filename):
```
Comment (da-woods) [inline Cython/Build/Dependencies.py:54]: Just remove this function?

File: Cython/Compiler/Main.py (hunk 1 above)
Comment (da-woods) [reply Cython/Compiler/Main.py:12]: Yes, I think so

File: Cython/Compiler/PyrexTypes.py [hunk 3]
```diff
@@ -8,10 +8,7 @@
 import hashlib
 import re
 
-try:
-    reduce
-except NameError:
-    from functools import reduce
+from functools import reduce
 from functools import partial
```
Comment (da-woods) [inline Cython/Compiler/PyrexTypes.py:12]:
```suggestion
from functools import partial, reduce
```

File: Cython/Compiler/Main.py (hunk 1 above)
Comment (matusvalo) [reply Cython/Compiler/Main.py:12]: Fixed.

File: Cython/Debugger/libpython.py [hunk 4]
```diff
@@ -1164,16 +1150,16 @@ def _unichr_is_printable(char):
     return unicodedata.category(char) not in ("C", "Z")
 
 if sys.maxunicode >= 0x10000:
```
Comment (scoder) [inline Cython/Debugger/libpython.py:1152]: This is always true now (PEP-393).

File: Cython/Compiler/ExprNodes.py [hunk 5]
```diff
@@ -53,23 +53,8 @@
...
-
-
-if sys.version_info[0] >= 3:
-    IS_PYTHON3 = True
-    _py_int_types = int
-else:
-    IS_PYTHON3 = False
-    _py_int_types = (int, long)
+any_string_type = (bytes, str)
+_py_int_types = int
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply Cython/Compiler/ExprNodes.py]: fixed.

File: Cython/Compiler/Optimize.py [hunk 6]
```diff
@@ -11,12 +10,9 @@
                Nodes=object, ExprNodes=object, PyrexTypes=object, Builtin=object,
                UtilNodes=object, _py_int_types=object)
 
-if sys.version_info[0] >= 3:
-    _py_int_types = int
-    _py_string_types = (bytes, str)
-else:
-    _py_int_types = (int, long)
-    _py_string_types = (bytes, unicode)
+_py_int_types = int
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply Cython/Compiler/Optimize.py]: fixed.

File: pyximport/pyximport.py [hunk 7]
```diff
@@ -1,11 +1,7 @@
 from __future__ import absolute_import
 import sys
 
-if sys.version_info < (3, 5):
-    # _pyximport3 module requires at least Python 3.5
-    from pyximport._pyximport2 import install, uninstall, show_docs
-else:
-    from pyximport._pyximport3 import install, uninstall, show_docs
+from pyximport._pyximport import install, uninstall, show_docs
```
[the remark this answers is not in the document]
Comment (matusvalo) [reply pyximport/pyximport.py]: done.

File: Cython/Debugger/libpython.py (hunk 4 above)
Comment (matusvalo) [reply Cython/Debugger/libpython.py]: fixed.

**Comment by scoder** [comment]
I recall that we took care to keep the history of `pyximport.py` when we split it. Maybe you can remove the old file and replace it with the Py3 version file directly?

**Comment by scoder** [comment]
Nice, thanks!

**Pull request merged**